scripts/validationAndPreparation.sh

```diff
@@ -1,45 +1,62 @@
 
-FILES=`lftp -e "set ssl:key-file ~/.ssh/id_rsa; cls; bye" sftp://josojo:@trusted-setup.staging.gnosisdev.com`
-if [[ -z "${NEWESTDATE}" ]]; then
-  NEWESTDATE=0
+if [[ -z "${DATE_OF_NEWEST_CONTRIBUTION}" ]]; then
+  DATE_OF_NEWEST_CONTRIBUTION=1
 fi
-if [[ -z "${TRUSTEDSETUPTURN}" ]]; then
-  TRUSTEDSETUPTURN=1
+
+if [[ -z "${TRUSTED_SETUP_TURN}" ]]; then
+  TRUSTED_SETUP_TURN=1
 fi
-unset NEWESTFILE
-for f in $FILES
-do
-	echo "Processing $f"
-	DATE=` lftp -e "set ssl:key-file ~/.ssh/id_rsa; cls -l --time-style=%FT%T $f/* --sort=date | head -1; bye" sftp://josojo:@trusted-setup.staging.gnosisdev.com | awk '{print $6}' | sed 's/[^0-9]*//g'`
-	echo "DATE is $DATE"
-	if [ $DATE -ge $NEWESTDATE ]; then
-		echo "found newer date"
-		NEWESTDATE=$DATE
-		NEWESTFILE=`lftp -e "set ssl:key-file ~/.ssh/id_rsa; cls -l --time-style=%FT%T $f/* --sort=date | head -1; bye" sftp://josojo:@trusted-setup.staging.gnosisdev.com | awk '{print $7}'`
-		echo "newFileName $NEWESTFILE"
-	fi	
-done
-
-if [[ !  -z "${NEWESTFILE}" ]]; then
+
+set -e 
+
+NEWEST_CONTRIBUTION=`lftp sftp://"$SSH_USER":@trusted-setup.staging.gnosisdev.com -e "set sftp:connect-program \"ssh -a -x -i ~/.ssh/$SSH_FILE\"; find -l | grep \"response$\" | sort -k4 | tail -1; bye"`
+NEWEST_CONTRIBUTION_DATE=`echo "$NEWEST_CONTRIBUTION" | awk '{print $4 $5}' | sed 's/[^0-9]*//g'`
+NEWEST_CONTRIBUTION_NAME=`echo "$NEWEST_CONTRIBUTION" | awk '{print $6}'`
+NEWEST_CONTRIBUTION_NAME=${NEWEST_CONTRIBUTION_NAME:2}
+if [ $NEWEST_CONTRIBUTION_DATE -gt $DATE_OF_NEWEST_CONTRIBUTION ]; then
+				
+	echo "current newest contribution is $NEWEST_CONTRIBUTION_NAME with the time $NEWEST_CONTRIBUTION_DATE"
+
+	#safe date of newest contribution so that files are not verified twice
+	export DATE_OF_NEWEST_CONTRIBUTION=$NEWEST_CONTRIBUTION_DATE #used for easy testing with source command
+	echo "export DATE_OF_NEWEST_CONTRIBUTION=$NEWEST_CONTRIBUTION_DATE " >> /root/project_env.sh #used for env export with cron job
+
+	#If a new contribution is found, do verification and preparation for next round
+	cd /app/
 	echo "starting download; this could take a while..."
-	sftp josojo@trusted-setup.staging.gnosisdev.com:${NEWESTFILE} ~/
+	$connect_to_sftp_server:$NEWEST_CONTRIBUTION_NAME /app/.
 
 	echo "verifying the submission; this could take a while..."
-	cargo run --release --bin verify_transform
-
+	if [[ ! -z "${CONSTRAINED}" ]]; then
+		cargo run --release --bin verify_transform_constrained
+	else
+		cargo run --release --bin verify_transform
+	fi
 
 	echo "uploading to ftp server and documentation; this could take a while..."
-	mv mv new_challenge challenge
-	mv response "response-$NEWESTFILE-$TRUSTEDSETUPTURN"
-
+	mv new_challenge challenge
+	mv response "response-$TRUSTED_SETUP_TURN"
+	
 	#upload new challenge file for next candiate
-	echo "put challenge" | sftp josojo@trusted-setup.staging.gnosisdev.com:josojo
+	echo "put challenge" | $connect_to_sftp_server:challenges
```

Comment:
maybe it would make sense to expose a helper method upload_to_server(src, dest)/download_from_server(src, dest)?